libbeat/beat/template.go

```diff
@@ -0,0 +1,67 @@
+package beat
+
+/*
+
+Load the index template. This replaces the manual running of
+curl -XPUT 'http://localhost:9200/_template/filebeat?pretty' -d@/etc/filebeat/filebeat.template.json
+
+
+TODO:
+ - Cleanup TODO's
+ - Should the template also be automatically loaded by logstash-input-beats?
+
+*/
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/outputs/elasticsearch"
+)
+
+var template *string
+
+func init() {
+	template = flag.String("template", "", "Path the index template")
+}
+
+func (beat *Beat) loadTemplate() error {
+
+	// TODO: Fin a way to check if flag was even set. If it was set but no path, default path should be used.
```

Comment:
Ah, right, I was hoping it sets the pointer to nil if the flag is not used, but that doesn't seem to be the case. Maybe we'd be better with having the setting in the configuration file? I find that more friendly to packaging, config management, etc.